cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -280,9 +283,27 @@ void DecisionTreeBase<T, L>::plant(
 
   total_temp_mem = tempmem->totalmem;
   MLCommon::TimerCPU timer;
-  grow_deep_tree(data, labels, rowids, n_sampled_rows, ncols,
-                 tree_params.max_features, dinfo.NLocalrows, sparsetree, treeid,
-                 tempmem);
+  if (tree_params.use_experimental_backend == true &&
+      tree_params.max_depth > 0 && tree_params.max_depth < 14 &&
+      tree_params.max_features == 1.0 &&
+      tree_params.split_algo == SPLIT_ALGO::GLOBAL_QUANTILE &&
+      tree_params.quantile_per_tree == false) {
```

Comment:
If use_experimental_backend is true, but some other condition (like split_algo or max_depth) forces us down the other path, would be really valuable to warn. Otherwise people will think they're using the new backend when they are not.

Reply:
+1

Reply:
Added bunch of warnings. If it full too verbose, all can be combined in one warning giving all the required conditions in one message.